lib/exercism/excerpt.rb

```diff
@@ -0,0 +1,26 @@
+class Excerpt
+  def initialize(text)
+    @html_text = text
+  end
+
+  def limit(length)
+    text[0..(length - 1)] + ellipsis(length)
+  end
+
+  private
+
+  def text
+    @_raw_text ||= begin
+      document = Nokogiri::HTML(@html_text)
+      remove_whitespace(document.text)
+    end
+  end
+
+  def remove_whitespace(string)
+    string.split("\n").map(&:strip).join(' ').strip
+  end
+
+  def ellipsis(length)
+    text.length < length ? '' : '...'
```

Comment:
@kytrinyx Yeah, the `&hellip;` gets escaped in the view. The only thing that comes to mind is to move the ellipsis logic to the view, but that's really ugly. :( Do you have any ideas on a better way to do this?